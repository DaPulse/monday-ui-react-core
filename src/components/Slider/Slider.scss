@import "../../styles/themes.scss";

.monday-slider {
  align-items: center;
  display: flex;
  max-width: 500px;
  min-width: 200px;
  width: 100%;

  &__prefix {
    font-size: 14px;
    margin-right: 8px;
  }

  &__postfix {
    font-size: 14px;
    margin-left: 8px;
  }

}