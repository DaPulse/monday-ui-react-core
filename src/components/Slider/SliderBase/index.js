export { default as SliderBase } from "./SliderBase";
export { useSliderValues } from "./SliderBaseHooks";
