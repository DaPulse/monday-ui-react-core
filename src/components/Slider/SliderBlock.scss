@import "../../styles/themes.scss";

.monday-slider {
  width: 100%;
  min-width: 100px;
  max-width: 500px;
}
