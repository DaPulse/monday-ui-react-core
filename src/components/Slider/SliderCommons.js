import cx from "classnames";

export const COMPONENT_ID = "slider";

// TODO: move to common constants?
export const SIZES_BASIC = Object.freeze({
  SMALL: "small",
  MEDIUM: "medium",
  LARGE: "large"
});

// TODO: move to common constants/utils?
// TODO: add tests
const BEM_PREFIX = "monday";
export function createBemHelper(block) {
  if (!block || block === "") {
    return () => "";
  }
  const blockClass = `${BEM_PREFIX}-${block}`;
  return function bem(element, modifiers = "", extraClasses = "") {
    const elClass = element !== "" ? `${blockClass}__${element}` : blockClass;
    const modClasses = cx(modifiers)
      .split(" ")
      .map(modClass => (modClass === "" ? "" : `${elClass}--${modClass}`));
    return cx(elClass, modClasses, extraClasses);
  };
}

export function ensureValueText(valueText, value, formatter) {
  if (valueText) {
    return valueText;
  }
  if (typeof value === "undefined") {
    return undefined;
  }
  if (typeof formatter !== "function") {
    return value;
  }
  return formatter(value);
}
