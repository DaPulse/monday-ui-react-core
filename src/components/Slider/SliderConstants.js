export const BEM_PREFIX = "monday";

export const COMPONENT_ID = "slider";

export const INFIX_KIND = {
  PREFIX: "prefix",
  POSTFIX: "postfix"
};

export const UPDATE_SLIDER_SIZE_DEBOUNCE = 200;