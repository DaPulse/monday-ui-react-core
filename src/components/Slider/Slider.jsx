import React, { forwardRef, useRef } from "react";
import PropTypes from "prop-types";
import { COLORS_BASIC, SIZES_BASIC } from "../../constants";
import useMergeRefs from "../../hooks/useMergeRefs";
import "./Slider.scss";
import { bem } from "./SliderCommons";
import { SliderProvider } from "./SliderContext";
import SliderBase from "./SliderBase/SliderBase";
import SliderInfix from "./SliderInfix";
import { createBemBlockHelper } from "../../helpers/bem-helper";

const Slider = forwardRef(
  (
    {
      ariaLabel,
      ariaLabeledBy,
      className,
      classNameBase,
      color,
      disabled,
      id,
      max,
      min,
      onChange,
      step,
      showValue,
      size,
      value,
      valueDefault,
      valueFormatter,
      valueText,
      // ------ Additional subcomponents' props
      indicateSelection,
      prefix,
      postfix
    },
    ref
  ) => {
    const componentRef = useRef(null);
    const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
    const consumerBem = createBemBlockHelper(classNameBase, { isConsume: true });
    function handleChange(newValue) {
      console.log("slider: handle change", newValue);
      if (typeof onChange === "function") {
        onChange(newValue);
      }
    }
    const providerProps = {
      ariaLabel,
      ariaLabeledBy,
      classNameBase,
      color,
      disabled,
      max,
      min,
      onChange,
      showValue,
      size,
      step,
      value,
      valueDefault,
      valueFormatter,
      valueText,
      infixOptions: {
        prefix,
        postfix,
        indicateSelection
      }
    };
    console.log("--- slider", { disabled });
    return (
      <SliderProvider {...providerProps}>
        <div ref={mergedRef} className={bem("", { disabled }, className)} id={id}>
          <SliderInfix kind="prefix" />
          <SliderBase className={consumerBem("base")} onChange={handleChange} />
          <SliderInfix kind="postfix" />
        </div>
      </SliderProvider>
    );
  }
);

Slider.sizes = SIZES_BASIC;
Slider.colors = COLORS_BASIC;

Slider.propTypes = {
  // ------ SliderBase props
  /**
   * Define a string that labels the current element (Slider)
   */
  ariaLabel: PropTypes.string,
  /**
   * ElementId of Node that have the text needed for labeling the current element (Slider)
   */
  ariaLabeledBy: PropTypes.string,
  /**
   * Custom `class name` to be added to the component-root-node
   */
  className: PropTypes.string,
  /**
   * Custom base of class names for add to component-root and internal-components nodes
   *  - check Dev Tools for found available Custom/Consumer Classes
   */
  classNameBase: PropTypes.string,
  /**
   * Color Mode (primary/positive/negative) of the component (Slider)
   */
  color: PropTypes.oneOf(Object.values(Slider.colors)),
  /**
   * If set to true, Component (Slider) will be disabled
   *  - impossible to change value of component (Slider)
   *  - visual changes (opacity)
   */
  disabled: PropTypes.bool,
  /**
   * Attribute `id` to be added to the component-root-node
   */
  id: PropTypes.string,
  /**
   * Max range value of the component (Slider)
   */
  max: PropTypes.number,
  /**
   * Min range value of the component (Slider)
   */
  min: PropTypes.number,
  /**
   * Optional onChange callback (for outer trigger or Controlled mode)
   * - required in Controlled Mode
   */
  onChange: PropTypes.func,
  /**
   * Step for Component (Slider) changes
   */
  step: PropTypes.number,
  /**
   * Always show `value` instead of Tooltip
   */
  showValue: PropTypes.bool,
  /**
   * Size small/medium/large of the component (Slider)
   */
  size: PropTypes.oneOf(Object.values(Slider.sizes)),
  /**
   * Current/selected value of the range of the Component (Slider)
   *   - should be used in Controlled Mode only
   */
  value: PropTypes.number,
  /**
   * Default `value` if value not specified
   *  - should be used in NON-Controlled mode to set initial Value
   */
  valueDefault: PropTypes.number,
  /**
   * Formatter function `value => formattedValue`
   * default formatter return `${value}%`
   */
  valueFormatter: PropTypes.func,
  /**
   * Text/presentation of current/selected value
   *  - should be used in Controlled Mode only
   */
  valueText: PropTypes.string,

  // ------ Additional subcomponents' props
  /**
   * Show selected from Slider range value
   */
  indicateSelection: PropTypes.bool,
  /**
   * Options for initial/start/prefix element, it can be one of:
   *  - Any Component (react component, node, text, number etc.)
   *  - Or it can be an object of options for Icons component (see Icon components props)
   *  - Or it can be an object for Label (Icon, Title - and other components)
   *  - Or it can be Render Props Function witch are getting value and valueText
   */
  prefix: PropTypes.oneOfType([
    PropTypes.shape({
      icon: PropTypes.string.isRequired
    }),
    PropTypes.shape({
      text: PropTypes.string.isRequired
    }),
    PropTypes.func,
    PropTypes.node
  ]),
  /**
   * Options for postfix/end/finishing element. Same as prefix element.
   */
  postfix: PropTypes.oneOfType([
    PropTypes.shape({
      icon: PropTypes.string.isRequired
    }),
    PropTypes.shape({
      text: PropTypes.string.isRequired
    }),
    PropTypes.func,
    PropTypes.node
  ])
};

Slider.defaultProps = {
  // ------ SliderBase props
  ariaLabel: undefined,
  ariaLabeledBy: undefined,
  className: "",
  classNameBase: "",
  color: undefined,
  disabled: false,
  id: undefined,
  max: 100,
  min: 0,
  onChange: (e) => {console.log('outer change', e)},
  step: 1,
  size: SIZES_BASIC.SMALL,
  showValue: false,
  value: undefined,
  valueDefault: 0,
  valueFormatter: value => `${value}%`,
  valueText: undefined,
  // ------ Additional subcomponents' props
  indicateSelection: false,
  prefix: undefined,
  postfix: undefined
};

export default Slider;
