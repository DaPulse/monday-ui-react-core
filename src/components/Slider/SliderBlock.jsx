import React, { forwardRef, useRef } from "react";
import PropTypes from "prop-types";
import useMergeRefs from "../../hooks/useMergeRefs";
import "./SliderBlock.scss";
import { bem } from "./SliderCommons";

const SliderBlock = forwardRef(({ children, className, disabled, id }, ref) => {
  const componentRef = useRef(null);
  const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
  console.log("slider: block", { children, className, id });
  return (
    <div ref={mergedRef} className={bem("", { disabled }, className)} id={id}>
      {children}
    </div>
  );
});

SliderBlock.propTypes = {
  /**
   * Consumer/Custom/Extra `class names` to be added to the Component's-Root-Node
   */
  className: PropTypes.string,
  /**
   * If set to true, Component will be disabled
   */
  disabled: PropTypes.bool,
  /**
   * Attribute `id` to be added to the Component's-Root-Node
   */
  id: PropTypes.string
};

SliderBlock.defaultProps = {
  className: undefined,
  disabled: false,
  id: undefined
};

export default SliderBlock;
