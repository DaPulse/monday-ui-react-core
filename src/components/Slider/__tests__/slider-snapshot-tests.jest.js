import React from "react";
import { cleanup, render, waitFor } from "@testing-library/react";
import { snapshotDiff } from "../../../../jest/utils";
import Slider from "../Slider";
import Sound from "../../Icon/Icons/components/Sound";
import Chips from "../../Chips/Chips";

async function waitForSliderRender(getByTestId, dataTestId) {
  const elThumb = getByTestId(dataTestId || "monday-slider__thumb-0");
  await waitFor(() => expect(elThumb).toBeTruthy());
}

async function renderSliderHelper(props, dataTestId) {
  const { asFragment, getByTestId } = render(<Slider {...props} />);
  await waitForSliderRender(getByTestId, dataTestId);
  return asFragment().firstChild;
}

describe("Slider Render", () => {
  let defaultRender;
  beforeAll(async () => {
    const { asFragment, getByTestId } = render(<Slider />);
    await waitForSliderRender(getByTestId);
    defaultRender = asFragment().firstChild;
    cleanup();
  });

  afterEach(() => {
    cleanup();
  });

  it("01. with empty props", async () => {
    expect(defaultRender).toMatchSnapshot();
  });

  it("02. with ariaLabel", async () => {
    const props = { ariaLabel: "My Slider" };
    const currentRender = await renderSliderHelper(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it("03. with ariaLabeledBy", async () => {
    const props = { ariaLabeledBy: "id-of-the-custom-label" };
    const currentRender = await renderSliderHelper(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it("04. with custom className and classNameBase", async () => {
    const props = { className: "slide-me", classNameBase: "slide-me" };
    const currentRender = await renderSliderHelper(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it("05. with id and data-testid", async () => {
    const props = { id: "my-slider", "data-testid": "my-slider" };
    const currentRender = await renderSliderHelper(props, "my-slider__thumb-0");
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it("06. with disabled", async () => {
    const props = { disabled: true };
    const currentRender = await renderSliderHelper(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it("07. with color", async () => {
    const props = { color: Slider.colors.POSITIVE };
    const currentRender = await renderSliderHelper(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it(`08. with size`, async () => {
    const props = { size: Slider.sizes.LARGE };
    const currentRender = await renderSliderHelper(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it(`09. with min, max, and step`, async () => {
    const props = { min: 10, max: 20, step: 2 };
    const currentRender = await renderSliderHelper(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it(`10. with showValue`, async () => {
    const props = { showValue: true };
    const currentRender = await renderSliderHelper(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it(`11. with valueFormatter`, async () => {
    const props = { valueFormatter: value => `--${value}GB--` };
    const currentRender = await renderSliderHelper(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it(`12. with value (controlled component)`, async () => {
    const props = { value: 12 };
    const currentRender = await renderSliderHelper(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it(`13. with value, valueText (controlled component)`, async () => {
    const props = { value: 15, valueText: "15kg" };
    const currentRender = await renderSliderHelper(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it(`14. with indicateSelection`, async () => {
    const props = { indicateSelection: true };
    const currentRender = await renderSliderHelper(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it(`15. with prefix (string) and postfix (Icon)`, async () => {
    const props = { prefix: "Vol", postfix: { icon: Sound } };
    const currentRender = await renderSliderHelper(props);
    expect(snapshotDiff(defaultRender, currentRender)).toMatchSnapshot();
  });

  it(`16. with prefix (custom) and postfix (render-props)`, async () => {
    const props = {
      prefix: <Chips label="Custom component" readOnly />,
      postfix: (value, valueText) => `Render Props result: ${valueText} (${value})`
    };
    const currentRender = await renderSliderHelper(props);
    expect(snapshotDiff(defaultRender, currentRender)).toMatchSnapshot();
  });

  it(`17. with isRange (ranged slider)`, async () => {
    const props = { isRange: true };
    const currentRender = await renderSliderHelper(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it(`18. ranged slider with options`, async () => {
    const props = { isRange: true, valueDefault: [20, 30], min: 10, max: 50, step: 2 };
    const currentRender = await renderSliderHelper(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });
});
