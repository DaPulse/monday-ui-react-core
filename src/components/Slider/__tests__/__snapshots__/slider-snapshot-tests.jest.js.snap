// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Render 01. with empty props 1`] = `
<div
  class="monday-slider"
  data-testid="monday-slider"
>
  <div
    class="monday-slider__base monday-slider__base--small"
    data-testid="monday-slider__base"
  >
    <div
      class="monday-slider__rail"
    >
      <div
        class="monday-slider__track"
      />
      <div
        class="monday-slider__filled-track"
        style="left: 0%; width: 0%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        aria-valuetext="0%"
        class="monday-slider__thumb monday-slider__thumb--index-0"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`Slider Render 02. with ariaLabel 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"ariaLabel":"My Slider"}

@@ --- --- @@
        />
        <div
          aria-disabled="false"
+         aria-label="My Slider"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
`;

exports[`Slider Render 03. with ariaLabelledby 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"ariaLabelledby":"id-of-the-custom-label"}

@@ --- --- @@
        />
        <div
          aria-disabled="false"
+         aria-labelledby="id-of-the-custom-label"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
`;

exports[`Slider Render 04. with custom className 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"className":"slide-me"}

@@ --- --- @@
  <div
-   class="monday-slider"
+   class="monday-slider slide-me"
    data-testid="monday-slider"
  >
    <div
`;

exports[`Slider Render 05. with id and data-testid 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"id":"my-slider","data-testid":"my-slider"}

@@ --- --- @@
  <div
    class="monday-slider"
-   data-testid="monday-slider"
+   data-testid="my-slider"
+   id="my-slider"
  >
    <div
      class="monday-slider__base monday-slider__base--small"
-     data-testid="monday-slider__base"
+     data-testid="my-slider__base"
    >
      <div
        class="monday-slider__rail"
@@ --- --- @@
          aria-valuenow="0"
          aria-valuetext="0%"
          class="monday-slider__thumb monday-slider__thumb--index-0"
-         data-testid="monday-slider__thumb-0"
+         data-testid="my-slider__thumb-0"
          role="slider"
          style="left: 0%;"
          tabindex="0"
`;

exports[`Slider Render 06. with disabled 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"disabled":true}

@@ --- --- @@
  <div
-   class="monday-slider"
+   class="monday-slider monday-slider--disabled"
    data-testid="monday-slider"
  >
    <div
-     class="monday-slider__base monday-slider__base--small"
+     class="monday-slider__base monday-slider__base--small monday-slider__base--disabled"
      data-testid="monday-slider__base"
    >
      <div
@@ --- --- @@
          style="left: 0%; width: 0%;"
        />
        <div
-         aria-disabled="false"
+         aria-disabled="true"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          aria-valuetext="0%"
-         class="monday-slider__thumb monday-slider__thumb--index-0"
+         class="monday-slider__thumb monday-slider__thumb--disabled monday-slider__thumb--index-0"
          data-testid="monday-slider__thumb-0"
          role="slider"
          style="left: 0%;"
-         tabindex="0"
+         tabindex="-1"
        />
      </div>
    </div>
`;

exports[`Slider Render 07. with color 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"color":"positive"}

@@ --- --- @@
    data-testid="monday-slider"
  >
    <div
-     class="monday-slider__base monday-slider__base--small"
+     class="monday-slider__base monday-slider__base--small monday-slider__base--positive"
      data-testid="monday-slider__base"
    >
      <div
`;

exports[`Slider Render 08. with size 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"size":"large"}

@@ --- --- @@
    data-testid="monday-slider"
  >
    <div
-     class="monday-slider__base monday-slider__base--small"
+     class="monday-slider__base monday-slider__base--large"
      data-testid="monday-slider__base"
    >
      <div
`;

exports[`Slider Render 09. with min, max, and step 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"min":10,"max":20,"step":2}

@@ --- --- @@
        />
        <div
          aria-disabled="false"
-         aria-valuemax="100"
-         aria-valuemin="0"
-         aria-valuenow="0"
-         aria-valuetext="0%"
+         aria-valuemax="20"
+         aria-valuemin="10"
+         aria-valuenow="10"
+         aria-valuetext="10%"
          class="monday-slider__thumb monday-slider__thumb--index-0"
          data-testid="monday-slider__thumb-0"
          role="slider"
`;

exports[`Slider Render 10. with showValue 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"showValue":true}

@@ --- --- @@
  <div
-   class="monday-slider"
+   class="monday-slider monday-slider--value-shown"
    data-testid="monday-slider"
  >
    <div
@@ --- --- @@
          role="slider"
          style="left: 0%;"
          tabindex="0"
-       />
+       >
+         <label
+           class="monday-slider__thumb-label"
+         >
+           0%
+         </label>
+       </div>
      </div>
    </div>
  </div>
`;

exports[`Slider Render 11. with valueFormatter 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {}

@@ --- --- @@
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
-         aria-valuetext="0%"
+         aria-valuetext="--0GB--"
          class="monday-slider__thumb monday-slider__thumb--index-0"
          data-testid="monday-slider__thumb-0"
          role="slider"
`;

exports[`Slider Render 12. with value (controlled component) 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"value":12}

@@ --- --- @@
        />
        <div
          class="monday-slider__filled-track"
-         style="left: 0%; width: 0%;"
+         style="left: 0%; width: 12%;"
        />
        <div
          aria-disabled="false"
          aria-valuemax="100"
          aria-valuemin="0"
-         aria-valuenow="0"
-         aria-valuetext="0%"
+         aria-valuenow="12"
+         aria-valuetext="12%"
          class="monday-slider__thumb monday-slider__thumb--index-0"
          data-testid="monday-slider__thumb-0"
          role="slider"
-         style="left: 0%;"
+         style="left: 12%;"
          tabindex="0"
        />
      </div>
`;

exports[`Slider Render 13. with value, valueText (controlled component) 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"value":15,"valueText":"15kg"}

@@ --- --- @@
        />
        <div
          class="monday-slider__filled-track"
-         style="left: 0%; width: 0%;"
+         style="left: 0%; width: 15%;"
        />
        <div
          aria-disabled="false"
          aria-valuemax="100"
          aria-valuemin="0"
-         aria-valuenow="0"
-         aria-valuetext="0%"
+         aria-valuenow="15"
+         aria-valuetext="15kg"
          class="monday-slider__thumb monday-slider__thumb--index-0"
          data-testid="monday-slider__thumb-0"
          role="slider"
-         style="left: 0%;"
+         style="left: 15%;"
          tabindex="0"
        />
      </div>
`;

exports[`Slider Render 14. with indicateSelection 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"indicateSelection":true}

@@ --- --- @@
        />
      </div>
    </div>
+   <div
+     class="monday-slider__postfix"
+   >
+     <span
+       class=""
+     >
+       <div
+         class="monday-style-label monday-style-label--kind-line monday-style-label--color-dark monday-style-label--with-animation"
+       >
+         <span>
+           0%
+         </span>
+         <span
+           class="monday-style-label__leg-wrapper"
+         />
+       </div>
+     </span>
+   </div>
  </div>
`;

exports[`Slider Render 15. with prefix (string) and postfix (Icon) 1`] = `
Snapshot Diff:
- Default Render
+ Current Render

@@ --- --- @@
    data-testid="monday-slider"
  >
    <div
+     class="monday-slider__prefix monday-slider__prefix--txt"
+   >
+     Vol
+   </div>
+   <div
      class="monday-slider__base monday-slider__base--small"
      data-testid="monday-slider__base"
    >
@@ --- --- @@
          style="left: 0%;"
          tabindex="0"
        />
+     </div>
    </div>
+   <div
+     class="monday-slider__postfix"
+   >
+     <svg
+       aria-hidden="true"
+       class="icon_component icon_component--no-focus-style"
+       fill="currentColor"
+       height="18"
+       viewBox="0 0 20 20"
+       width="18"
+     >
+       <path
+         clip-rule="evenodd"
+         d="M11.7497 3.25772C12.113 3.22596 12.4792 3.2925 12.8078 3.45138 13.1363 3.6106 13.4168 3.8572 13.6137 4.16598 13.8109 4.47502 13.9165 4.83337 13.9163 5.20068V14.8C13.9163 15.1674 13.8104 15.5258 13.6129 15.8347 13.4157 16.1434 13.1357 16.3895 12.8069 16.5485 12.4784 16.7074 12.1122 16.774 11.749 16.7423 11.3962 16.7116 11.0568 16.589 10.7666 16.3856L10.7408 16.3671C10.7493 16.3733 10.758 16.3795 10.7666 16.3856L5.50977 13.1501H5.72208V12.4001L6.5 12.5 6.47208 11.981 11.5743 15.1213C11.5904 15.1312 11.6062 15.1418 11.6215 15.1529 11.6944 15.2058 11.7836 15.2396 11.8793 15.248 11.975 15.2563 12.0703 15.2385 12.1539 15.1981 12.2373 15.1578 12.304 15.0975 12.349 15.027 12.3938 14.9569 12.4163 14.8783 12.4163 14.8V5.20019C12.4163 5.12191 12.3939 5.04289 12.3491 4.9727 12.3043 4.90237 12.2379 4.84213 12.1547 4.80173 12.0711 4.76139 11.9759 4.74366 11.8803 4.75202 11.7846 4.76039 11.6954 4.79425 11.6225 4.84718 11.6071 4.85837 11.5913 4.86897 11.575 4.87895L6.47208 8.01925V7.60015L5.72208 7.60015V6.85015H5.5098L10.7674 3.61469C10.7587 3.6208 10.75 3.62699 10.7414 3.63325L10.7674 3.61469C11.0575 3.41116 11.3969 3.28856 11.7497 3.25772zM4.97208 8.35015V11.6501H3.24069C3.10263 11.6501 2.97505 11.5968 2.8848 11.5095 2.7954 11.423 2.75 11.3112 2.75 11.2001V8.80013C2.75 8.68902 2.7954 8.57718 2.8848 8.49072 2.97505 8.40343 3.10263 8.35015 3.24069 8.35015H4.97208zM5.5098 6.85015H3.24069C2.72065 6.85015 2.21708 7.04972 1.84198 7.41251 1.46603 7.77613 1.25 8.27473 1.25 8.80013V11.2001C1.25 11.7255 1.46603 12.2241 1.84198 12.5877 2.21708 12.9505 2.72066 13.1501 3.24069 13.1501H5.50977 5.72208V12.4001L6.5 12.5 6.47208 11.981V8.01925 7.60015L5.72208 7.60015V6.85015H5.5098zM16.2911 7.52882C16.5414 7.19876 17.0118 7.13408 17.3419 7.38435 17.778 7.71509 18.1329 8.13906 18.3777 8.62555 18.6225 9.11215 18.7504 9.6476 18.7504 10.1911 18.7504 10.7347 18.6225 11.2701 18.3777 11.7567 18.1329 12.2432 17.778 12.6672 17.3419 12.9979 17.0118 13.2482 16.5414 13.1835 16.2911 12.8534 16.0408 12.5234 16.1055 12.0529 16.4356 11.8027 16.6921 11.6082 16.8974 11.3614 17.0377 11.0825 17.1781 10.8036 17.2504 10.4989 17.2504 10.1911 17.2504 9.88337 17.1781 9.57866 17.0377 9.29981 16.8974 9.02084 16.6921 8.7741 16.4356 8.5796 16.1055 8.32933 16.0408 7.85888 16.2911 7.52882z"
+         fill="currentColor"
+         fill-rule="evenodd"
+       />
+     </svg>
    </div>
  </div>
`;

exports[`Slider Render 16. with prefix (custom) and postfix (render-props) 1`] = `
Snapshot Diff:
- Default Render
+ Current Render

@@ --- --- @@
    data-testid="monday-slider"
  >
    <div
+     class="monday-slider__prefix"
+   >
+     <div
+       class="chips--wrapper"
+       id=""
+     >
+       <div
+         class="label"
+       >
+         Custom component
+       </div>
+     </div>
+   </div>
+   <div
      class="monday-slider__base monday-slider__base--small"
      data-testid="monday-slider__base"
    >
@@ --- --- @@
          tabindex="0"
        />
      </div>
+   </div>
+   <div
+     class="monday-slider__postfix"
+   >
+     Render Props result: 0% (0)
    </div>
  </div>
`;

exports[`Slider Render 17. with isRange (ranged slider) 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"isRange":true}

@@ --- --- @@
        />
        <div
          class="monday-slider__filled-track"
-         style="left: 0%; width: 0%;"
+         style="left: 0%; width: 100%;"
        />
        <div
          aria-disabled="false"
@@ --- --- @@
          data-testid="monday-slider__thumb-0"
          role="slider"
          style="left: 0%;"
+         tabindex="0"
+       />
+       <div
+         aria-disabled="false"
+         aria-valuemax="100"
+         aria-valuemin="0"
+         aria-valuenow="100"
+         aria-valuetext="100%"
+         class="monday-slider__thumb monday-slider__thumb--index-1"
+         data-testid="monday-slider__thumb-1"
+         role="slider"
+         style="left: 100%;"
          tabindex="0"
        />
      </div>
`;

exports[`Slider Render 18. ranged slider with options 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"isRange":true,"valueDefault":[20,30],"min":10,"max":50,"step":2}

@@ --- --- @@
        />
        <div
          class="monday-slider__filled-track"
-         style="left: 0%; width: 0%;"
+         style="left: 25%; width: 25%;"
        />
        <div
          aria-disabled="false"
-         aria-valuemax="100"
-         aria-valuemin="0"
-         aria-valuenow="0"
-         aria-valuetext="0%"
+         aria-valuemax="50"
+         aria-valuemin="10"
+         aria-valuenow="20"
+         aria-valuetext="20%"
          class="monday-slider__thumb monday-slider__thumb--index-0"
          data-testid="monday-slider__thumb-0"
          role="slider"
-         style="left: 0%;"
+         style="left: 25%;"
+         tabindex="0"
+       />
+       <div
+         aria-disabled="false"
+         aria-valuemax="50"
+         aria-valuemin="10"
+         aria-valuenow="30"
+         aria-valuetext="30%"
+         class="monday-slider__thumb monday-slider__thumb--index-1"
+         data-testid="monday-slider__thumb-1"
+         role="slider"
+         style="left: 50%;"
          tabindex="0"
        />
      </div>
`;
